flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvReaderFormat.java

```diff
@@ -155,16 +169,34 @@
     private static final class Reader<R, T> implements StreamFormat.Reader<T> {
         private final MappingIterator<R> iterator;
         private final Converter<R, T, Void> converter;
+        private final boolean ignoreParseErrors;
 
-        public Reader(MappingIterator<R> iterator, Converter<R, T, Void> converter) {
+        public Reader(
+                MappingIterator<R> iterator,
+                Converter<R, T, Void> converter,
+                boolean ignoreParseErrors) {
             this.iterator = checkNotNull(iterator);
             this.converter = checkNotNull(converter);
+            this.ignoreParseErrors = ignoreParseErrors;
         }
 
         @Nullable
         @Override
         public T read() throws IOException {
-            return iterator.hasNext() ? converter.convert(iterator.next(), null) : null;
+            while (true) {
```

Comment:
Why do you now use a while loop?

Reply:
An arbitrary number of lines can be malformed. Let's say this is a single CSV file
```
a,b,c    // header
1,2,3    // expected
1,          // malformed
1,2,3,4 // malformed
5,6,7   // expected
```
Should produce [1,2,3] and [5,6,7] and ignore the rest. The number of malformed elements is unknown.

Reply:
I see then I'd propose to rewrite the loop to 

```java
while (iterator.hasNext()) {
    try {
        R nextElement = iterator.next();
        return converter.convert(nextElement, 
    } catch (....) {
    }
}
return null;
```